src/components/User.vue

```diff
@@ -1,38 +1,141 @@
-<script setup lang="ts">
-  import { createApp } from 'vue';
+<script lang="ts">
+import { defineComponent } from 'vue';
+import Search from './Search.vue';
 
+interface UserInfo {
+  avatar_url: string;
+  name: string | null;
+  login: string;
+  created_at: string;
+  bio: string | null;
+  public_repos: number;
+  followers: number;
+  following: number;
+  location: string | null;
+  blog: string | null;
+  twitter_username: string | null;
+  company: string | null;
+}
+
+export default defineComponent({
+  components: {
+    Search,
+  },
+  data() {
+    return {
+      username: 'octocat',
+      userInfo: null as UserInfo | null,
+      error: null as string | null,
+    };
+  },
+  methods: {
+    async fetchUserInfo() {
+      try {
+        const response = await fetch(`https://api.github.com/users/${this.username}`);
+        if (!response.ok) {
+          this.error = 'User not found';
+          this.userInfo = null;
+          return;
+        }
+        this.error = null;
+        this.userInfo = await response.json();
+        console.log(this.userInfo);
+      } catch (err) {
+        this.error = 'Error fetching user information';
+        this.userInfo = null;
+      }
+    },
+    updateUsername(newUsername: string) {
```

Comment:
The `updateUsername` method is calling `fetchUserInfo` whenever the `username` changes, which could result in multiple HTTP requests being made in a short amount of time. It's better to use a debounce or throttle function to limit the rate at which the method is called.